src/core/services.py

```diff
@@ -508,3 +511,73 @@ def queue_job(project_id: int, job_id: int, session: Session) -> None:
         logger.error(
             f"Cannot queue job {job_id}, it's of status {job.status()}."
         )
+
+
+def get_directory_listing(path: str) -> Dict[str, Any]:
+    """Get contents found in a given directory. Does not search recursivly.
+
+    Parameters
+    ----------
+    path    :   str
+        Path to the folder one wants to get listing from.
+
+    Raises
+    ------
+    NotADirectoryError
+        When the given path is a file and not a directory.
+
+    FileNotFoundError
+        Directory cannot be found at a given path.
+
+    Returns
+    -------
+    Dict
+        jsTree formatted json containing information about root node at path.
+    """
+    root_node: Dict[str, Any] = dict()
+    child_list = []
+
+    if isfile(path):
```

Comment:
Is this necessary when you are checking that you're working on a dir below?

Reply:
Yes, the one bellow checks if the path is valid. This one checks if it is a file, and not a directory. `os.walk(path)` throws an error if this a file.

Reply:
`isdir` simply returns `True` if `path` is an existing directory, which means it should also be valid for `os.walk`.

Reply:
```
if isfile(path): # Returns FALSE if it is an invalid path (it is not a file)
    raise NotADirectoryError("Path must be a directory.")

if not isdir(path): # If it is NOT a directory (invalid path)
    raise FileNotFoundError("Directory at '{}' was not found.".format(path))
````

Reply:
Having just one check `isdir` wouldnt make it possible to differentiate the type of error. That way it would throw "file not found" even if the path is valid, but pointing to a file.